packages/gatsby/src/utils/page-data.ts

```diff
@@ -162,7 +163,7 @@ export async function flush(): Promise<void> {
 
         if (hasFlag(query.dirty, FLAG_DIRTY_NEW_PAGE)) {
           // query results are not written yet
-          continue
+          cb(null, true)
```

Comment:
@pieh any idea why this isn't working?

Reply:
Should you also add `return` after this line (as we want to skip the remaining part of this function)?

Reply:
What Vlad said, or keep part of the loop as-is now, and only make async part of the loop use queue - so change https://github.com/gatsbyjs/gatsby/blob/bca9518abd0f3f4560c25b24fda04955baaf7528/packages/gatsby/src/utils/page-data.ts#L172-L185

to push to queue and queue handler only does that part

Reply:
🤦 yes, returning helps